src/osd/PG.cc

```diff
@@ -3178,6 +3177,13 @@ bool PG::sched_scrub()
       return false;
   }
 
+  if (time_for_deep 
+      && get_pgbackend()->auto_repair_supported()
+      && cct->_conf->osd_scrub_auto_repair) {
```

Comment:
Should also test that !scrubber.must_scrub here.  Now if scrubber.auto_repair is set we know that this could not have been an initiated must_scrub.